### PR TITLE
Introduce basic T2 topology

### PR DESCRIPTION
**Pull request by shubav**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?

Need a basic topology against which T2 test cases can be standardized and executed.

#### How did you do it?

The new T2 topology, as pictured in the README.testbed.Topology.md file, has:
3 line cards - dut_index 0, 1, and 2
1 supervisor card - dut_index 3
Each line card has 32 VMs.

BGP info:
DUT ASN: 65100
Linecard 1 (dut_index 0) has 32 eBGP peers, each with the same ASN 65200
Linecard 2 (dut_index 1) has 32 eBGP peers, each with a different ASN (65001-65132)
Linecard 3 (dut_index 2) has 32 eBGP peers, each with a different ASN (65033-65064)

#### How did you verify/test it?

Tests are not yet deployed. Hence, not applicable.

### CODE REVIEW

**Review by yxieca — APPROVED** [review]

**Comment by yxieca** [comment]
retest vsimage please

File: ansible/vars/topo_t2.yml [hunk 1]
```diff
@@ -0,0 +1,2325 @@
...
+    interfaces:
+      Loopback0:
+        ipv4: 100.1.0.1/32
+        ipv6: 2064:100::1/128
+      Ethernet1:
+        ipv4: 10.0.0.1/31
+        ipv6: FC00:2/126
+    bp_interface:
+      ipv4: 10.10.246.1
+      ipv6: fc0a::2/64
```
Comment (wangxin) [inline ansible/vars/topo_t2.yml:424]: Generally T2 and T3 are connected by PortChannels. Probably we can design the topology like the t1-lag topo?
* Use LAG between T2 and 8 T3 VMs
* Use single link between T2 and 16 T3 VMs.

Totally 24 T3 VMs. 

From T2 to T1, we can use similar design, 8 LAGs to 8 T1 VMs. 16 links to 16 T1 VMs. Totally 24 T1 VMs.

Benefits of the proposal:
* The required footprint would be smaller
* Both LAG and single link can be covered

File: ansible/vars/topo_t2.yml [hunk 2]
```diff
@@ -0,0 +1,2325 @@
...
+configuration_properties:
+  common:
+    dut_asn: 65100
+    dut_type: LeafRouter
+    nhipv4: 10.10.246.254
+    nhipv6: FC0A::FF
+  spine:
+    swrole: spine
+  tor:
+    swrole: tor
```
[the remark this answers is not in the document]
Comment (shubav) [reply ansible/vars/topo_t2.yml]: Removed in the latest commit, Xin. Thanks.

File: ansible/vars/topo_t2.yml (hunk 1 above)
Comment (shubav) [reply ansible/vars/topo_t2.yml:355]: Modified the testbed diagram, the testbed readme and the topo-t2 yml file as per your inputs. Thanks again

File: ansible/vars/topo_t2.yml [hunk 3]
```diff
@@ -0,0 +1,1867 @@
...
+    ARISTA01T3:
+      vlans:
+        - 0.0@0
+        - 0.1@1
+      vm_offset: 0
+    ARISTA03T3:
+      vlans:
+        - 0.2@2
+        - 0.3@3
+      vm_offset: 2
```
Comment (wangxin) [inline ansible/vars/topo_t2.yml:17]: The vm_offset is used to map the VM in topology to server VM names, like (ARISTA01T3 -> VM0300).
If there is gap in vm_offset number, then ARISTA03T3 would be mapped to VM0302. Then VM0301 would not be used.
Suggest to make the vm_offset continuous without gap.

File: ansible/vars/topo_t2.yml [hunk 4]
```diff
@@ -0,0 +1,1867 @@
...
+    interfaces:
+      Loopback0:
+        ipv4: 100.1.0.1/32
+        ipv6: 2064:100::1/128
+      Ethernet1:
+        lacp: 1
+      Ethernet2:
+        lacp: 1
+      Port-Channel1:
+        ipv4: 20.0.0.1/31
```
Comment (wangxin) [inline ansible/vars/topo_t2.yml:351]: Why assign PortChannel IP address 20.0.0.x? In other topologies, IP address 10.0.0.x is used.

Comment (sanmalho-git) [reply ansible/vars/topo_t2.yml:351]: Currently in our SONiC Chassis, the subnet 10.0.0.0/24 is used for backplane (midplane) connectivity between the cards (used for synchronization of VOQ related databases as well). Thus the default address scheme conflicts with the midplane address.  Will ask developers if the midplane address can be configurable.

File: ansible/vars/topo_t2.yml (hunk 3 above)
Comment (sanmalho-git) [reply ansible/vars/topo_t2.yml:17]: Agreed - will make the change once we get a resolution on the 20.0.0.x vs 10.0.0.x being used.

Comment (shubav) [reply ansible/vars/topo_t2.yml]: Committed the changes

File: ansible/vars/topo_t2.yml (hunk 4 above)
Comment (shubav) [reply ansible/vars/topo_t2.yml]: Committed the changes

**Review by wangxin — APPROVED** [review]

File: docs/testbed/README.testbed.Topology.md [hunk 5]
```diff
@@ -23,6 +23,13 @@
 
 ## Current topologies
 
+### t2
```
Comment (saravanansv) [inline docs/testbed/README.testbed.Topology.md:26]: Looks like t2 is assumed to be 1 supe, 3max LCs accommodating chassis and can't support any more LCs right other than 3 right ?
Might be worth mentioning it here.

Comment (shubav) [reply docs/testbed/README.testbed.Topology.md:26]: Actually, that's the base topology. We wanted to give an example of 1 upstream and 2 downstream line cards. Just as we can extend current t0.yml to make t0-52.yml, t0-112.yml, etc, we can extend this to add a fully populated sonic chassis too. Many infra PRs mentioned in the chassis subgroup tracker were put in place precisely to support this flexibility and line card additions.

**Pull request merged**